lib/src/main/java/org/team100/lib/util/CameraAngles.java

```diff
@@ -1,112 +1,29 @@
-            double xOffset,
-            double yOffset) {
-        m_downwardAngleDegrees = downwardAngleDegrees;
-        m_horzFOVDegrees = horzFOVDegrees;
-        m_vertFOVDegrees = vertFOVDegrees;
-        m_horzResolution = horzResolution;
-        m_vertResolution = vertResolution;
-        m_cameraHeightMeters = cameraHeightMeters;
-        m_xOffset = xOffset;
-        m_yOffset = yOffset;
-    }
-
-    /**
-     * Creates a camera angles class with no peramters, this is purely for testing
-     * and sim
-     */
-    public CameraAngles() {
-        m_downwardAngleDegrees = 0;
-        m_horzFOVDegrees = 0;
-        m_vertFOVDegrees = 0;
-        m_horzResolution = 0;
-        m_vertResolution = 0;
-        m_cameraHeightMeters = 0;
-        m_xOffset = 0;
-        m_yOffset = 0;
-    }
-
-    /**
-     * @return A robot relative translational x value of an object in a camera in
-     *         meters
-     */
-    public double getX(double vertPixels) {
-        double x = -1.0 * m_cameraHeightMeters
-                * Math.tan(vertPixels * (Math.toRadians(m_vertFOVDegrees / m_vertResolution))
-                        + Math.toRadians(90 - m_vertFOVDegrees / 2 - m_downwardAngleDegrees));
-        return x - m_xOffset;
+            Transform3d cameraInRobotCoordinates) {
+        m_cameraInRobotCoordinates = cameraInRobotCoordinates;
     }
 
     /**
+     * 0 should be center of FOV
+     * 
      * @return A robot relative translational y value of an object in a camera in
      *         meters
      */
-    public double getY(double horzPixels, double vertPixels) {
-        double x = getX(vertPixels);
-        double y = -1.0 * x
-                * Math.tan(Math.toRadians(m_horzFOVDegrees) * (horzPixels
-                        - m_horzResolution / 2) / m_horzResolution);
-        return y - m_yOffset;
-    }
-
-    /**
-     * @return A camera vertical pixel measurement given a translational x value
-     */
-    public double getInverseX(double forwardMeters) {
-        double x = (m_vertResolution * (Math.atan((forwardMeters + m_xOffset) / m_cameraHeightMeters)
-                - Math.toRadians(90 - m_vertFOVDegrees/2 - m_downwardAngleDegrees)))
-                / Math.toRadians(m_vertFOVDegrees);
-        return x;
-    }
-
-    /**
-     * @return A camera horizontal pixel measurement given an x and y translational value, 0 is 416 pixels, or centerscreen
-     */
-    public double getInverseY(double forwardMeters, double sideMeters) {
-        double y = m_horzResolution * (Math.atan((sideMeters+m_yOffset)/(forwardMeters+m_xOffset))/Math.toRadians(m_horzFOVDegrees)) + m_horzResolution/2;
-        return y;
-    }
-
-    /**
-     * @return A robot relative angle in radians to the note
-     */
-    public double getAngle(double horzPixels, double vertPixels) {
-        return Math.atan2(getY(horzPixels, vertPixels), getX(vertPixels));
+    public Translation2d getTranslation2d(Rotation3d cameraObject) {
```

Comment:
wpi supplies methods for composing 3d geometries, and you should use them.  so if you have camera in robot coordinates, and a rotation in the camera coordinates, then to get the rotation in robot coordinates you just say cameraInRobotCoordinates.plus(new Transform3d(new Translation3d(), rotationInCameraCoordinates);

also i think it would be more clear if you made the calculation more physically meaningful.  you're calculating the complement of the tilt angle, i.e. the angle between the vertical and the note, so just do that.  as it's written, it seems like you're calculating a different (large) angle that happens to have the same tangent as the one you really want.  it would be good to draw a picture of what you're doing.

also this can just be a static method in PoseEstimationHelper,, you don't need this class at all.